src/engine/renderer/tr_image.cpp

```diff
@@ -143,6 +143,417 @@ void GL_TextureMode( const char *string )
 	}
 }
 
+int GetFormatPixelSize( GLenum internalFormat )
+{
+
+	int s8 = 8;
+	int s16 = 16;
+	int i8 = 8;
+	int i16 = 16;
+	int i32 = 32;
+	int ui2= 2;
+	int ui8 = 8;
+	int ui10 = 10;
+	int ui16 = 16;
+	int ui32 = 32;
+	int f10 = 10;
+	int f11 = 11;
+	int f16 = 16;
+	int f32 = 32;
+
+	switch ( internalFormat )
+	{
+		/* Sized formats
+
+		Values described as:
+			red + green + blue + alpha + shared
+		borrowed from:
+			https://www.khronos.org/opengl/wiki/GLAPI/glTexStorage3D */
+
+		case GL_R8:
+			return 8;
+		case GL_R8_SNORM:
+			return s8;
+		case GL_R16:
+			return 16;
+		case GL_R16_SNORM:
+			return s16;
+		case GL_RG8:
+			return 8 + 8;
+		case GL_RG8_SNORM:
+			return s8 + s8;
+		case GL_RG16:
+			return 16 + 16;
+		case GL_RG16_SNORM:
+			return s16 + s16;
+		case GL_R3_G3_B2:
+			return 3 + 3 + 2;
+		case GL_RGB4:
+			return 4 + 4 + 4;
+		case GL_RGB5:
+			return 5 + 5 + 5;
+		case GL_RGB8:
+			return 8 + 8 + 8;
+		case GL_RGB8_SNORM:
+			return s8 + s8 + s8;
+		case GL_RGB10:
```

Comment:
Are you sure all these ones which are not divisible by 8 actually take up a non-integer number of bytes?

Probably the engine can't actually load any of the ones like that, so one option would be to just remove ones which are too much work to figure out.

Reply:
Values taken from https://www.khronos.org/opengl/wiki/GLAPI/glTexImage2D

I think for such situation it's better to be exhaustive even if it's unused: if we start to remove some, it will be hard to know if something is missing and what.

And yes, there is formats with non-integer numbers of bytes. It's even worse with ASTC since pixel size is a float.

Reply:
Now the code lives in its own file. I've made it as exhaustive as I could like if it was a repurposable library, even if we don't use every format on earth.